pallets/data-preservers/src/lib.rs

```diff
@@ -130,6 +172,84 @@ pub mod pallet {
         ValueQuery,
     >;
 
+    #[pallet::storage]
+    pub type Profiles<T: Config> =
+        StorageMap<_, Blake2_128Concat, ProfileId, RegisteredProfile<T>, OptionQuery>;
+
+    #[pallet::storage]
+    pub type NextProfileId<T: Config> = StorageValue<_, ProfileId, ValueQuery>;
+
+    /// Balance used by this pallet
+    pub type BalanceOf<T> =
+        <<T as Config>::Currency as Inspect<<T as frame_system::Config>::AccountId>>::Balance;
+
+    pub type ProfileId = u64;
+
+    /// Data preserver profile.
+    #[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+    #[derive(
+        RuntimeDebugNoBound, PartialEqNoBound, EqNoBound, Encode, Decode, CloneNoBound, TypeInfo,
+    )]
+    #[scale_info(skip_type_params(T))]
+    pub struct Profile<T: Config> {
+        pub url: BoundedVec<u8, T::MaxBootNodeUrlLen>,
+        pub limited_to_para_ids: Option<BoundedVec<ParaId, T::MaxParaIdsVecLen>>,
```

Comment:
It would also be interesting to allow blacklisting some para ids, I imagine the case of a chain that is very active so this data preserver doesn't want to handle all the load.

Reply:
so you want to have the choice between whitelisting and blacklisting?

Reply:
I guess it's not a bad idea and it does not cost much to add it

Reply:
Dont forget to update the deposit as we have new bytes now

Reply:
Deposit is computed from size so it should already take into account the change.